### PR TITLE
Add support for default issues

### LINKED ISSUE #12: Add Support for Default Issues
It would be useful to support the notion that we'd like to add some default prefix if no issue number was found

### PR DESCRIPTION
**Pull request by RobertLD**
This PR adds an additional parameter that lets a use specify a default issue prefix if one is not found by the provided regex matching.

There are some other minor syntax changes using the walrus operator for ease of readability 

fixes: #12

### CODE REVIEW

**Comment by avilaton** [comment]
just merged an earlier contribution, please have another look at this one

**Comment by RobertLD** [comment]
> just merged an earlier contribution, please have another look at this one

I've addressed the merge conflicts, feel free to review again at your leisure

File: add_msg_issue_prefix_hook/add_msg_issue_prefix.py [hunk 1]
```diff
@@ -74,12 +73,18 @@ def main():
...
+        help="Default prefix if no issue is found",
+    )
 
     args = parser.parse_args()
     commit_msg_filepath = args.commit_msg_filepath
     template = args.template
+    default = args.default
     insert_after = re.compile(args.insert_after)
-    pattern = re.compile(args.pattern)
+    pattern = re.compile(args.insert_after)
```
Comment (ivysochyn) [inline add_msg_issue_prefix_hook/add_msg_issue_prefix.py]: This overrides `pattern` argument which defines regex for finding issue number.

File: add_msg_issue_prefix_hook/add_msg_issue_prefix.py [hunk 2]
```diff
@@ -89,12 +94,10 @@ def main():
     except Exception as e:
         print(e)
 
-    result = get_ticket_id_from_branch_name(pattern, branch)
-    issue_number = ""
-
-    if result:
+   if result := get_ticket_id_from_branch_name(pattern, branch):
```
Comment (ivysochyn) [inline add_msg_issue_prefix_hook/add_msg_issue_prefix.py]: Indentation.

**Review by ivysochyn — CHANGES_REQUESTED** [review]

File: add_msg_issue_prefix_hook/add_msg_issue_prefix.py (hunk 1 above)
Comment (RobertLD) [reply add_msg_issue_prefix_hook/add_msg_issue_prefix.py]: Doh Embarrassingly bad merge resolution, resolved

**Review by ivysochyn — APPROVED** [review]

**Pull request merged**